src/components/media/controls/videoControls.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react'
+
+import type { MediaControls } from './commonControls'
+import { MediaIconButton } from './mediaIconButton'
+
+interface FullscreenToggleProps {
+  element: HTMLElement
+}
+
+export function PictureInPictureToggle(props: MediaControls) {
+  const [isPictureInPicture, setIsPictureInPicture] = useState(
+    !!document.pictureInPictureElement
+  )
+
+  const videoElement = props.mediaElement as HTMLVideoElement
+
+  const action = isPictureInPicture
+    ? 'pictureInPictureExit'
+    : 'pictureInPicture'
+
+  function handlePictureInPicture() {
+    if (isPictureInPicture) {
+      document.exitPictureInPicture()
+    } else {
+      videoElement.requestPictureInPicture()
+    }
+  }
```

Comment:
Both `document.exitPictureInPicture()` and `videoElement.requestPictureInPicture()` return Promise. Use that instead of adding event handlers below.

Reply:
I've left the `onleavepictureinpicture` handler for when users exit from the pip window.